tests/web/url.py

```diff
@@ -141,6 +169,14 @@ def test_cyrillic_path(self):
         self.assertEqual(url1.path, '/%D1%82%D0%B5%D1%81%D1%82')
         self.assertEqual(url1.path, url2.path)
 
+    def test_empty_fragment(self):
+        self.assertEqual(URL.from_url('/').fragment, None)
+        self.assertEqual(URL.from_url('/#').fragment, '')
+
+        self.assertEqual(URL('/', fragment=None), '/')
+        self.assertEqual(URL('/', fragment='') ,'/#')
```

Comment:
Seems ok to me. It's easy to ignore the difference with test like `if url.fragment`.